### PR TITLE
boards: weact: stm32g474 core board

### PR DESCRIPTION
**Pull request by ankayca**
add weact stm32g474 core board support and documentation.

Signed-off-by: Ahmet Adin <ankayca2121@gmail.com>

### CODE REVIEW

File: boards/weact/stm32g474_core/weact_stm32g474_core.dts [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+	aliases {
+		led0 = &led_0;
+		mcuboot-led0 = &led_0;
+		sw0 = &user_button;
+		watchdog0 = &iwdg;
+		die-temp0 = &die_temp;
+		volt-sensor0 = &vref;
+		volt-sensor1 = &vbat;
+	};
+
```
Comment (nordicjm) [inline boards/weact/stm32g474_core/weact_stm32g474_core.dts:53]:
```suggestion
```

File: boards/weact/stm32g474_core/weact_stm32g474_core.dts [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+	};
+};
+&lpuart1 {
+	pinctrl-0 = <&lpuart1_tx_pa2 &lpuart1_rx_pa3>;
+	pinctrl-1 = <&analog_pa2 &analog_pa3>;
+	pinctrl-names = "default", "sleep";
+	current-speed = <115200>;
+	status = "okay";
+};
+&clk_lsi {
```
Comment (nordicjm) [inline boards/weact/stm32g474_core/weact_stm32g474_core.dts:74]: add newline gaps

File: boards/weact/stm32g474_core/weact_stm32g474_core_defconfig [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#
+# Copyright (c) 2024 Ahmet Adin.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+
```
Comment (nordicjm) [inline boards/weact/stm32g474_core/weact_stm32g474_core_defconfig:7]:
```suggestion
```

File: boards/weact/stm32g474_core/doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+| SB3/SB4       | Open    | Connect PA9/PA10 to CC routing               |
++---------------+---------+-----------------------------------------------+
+| SB5/SB6       | Open    | Connect PB6/PB4 to CC routing                |
++---------------+---------+-----------------------------------------------+
+| SB7/SB8       | Closed  | Connect CC1/CC2 to pull-down resistors       |
++---------------+---------+-----------------------------------------------+
+
+Supported Hardware Features
+--------------------------
+
```
Comment (nordicjm) [inline boards/weact/stm32g474_core/doc/index.rst:25]: Add a line similar to https://github.com/zephyrproject-rtos/zephyr/commit/887f210153b8a50e417232582762167d914c4033#diff-96c74ad50824463df0bd48ff926ecd261e62399a80a09226f40891629f2ea148R25

**Review by nordicjm — CHANGES_REQUESTED** [review]
commit title needs fixing, no `fix` or brackets, add sign off

File: boards/weact/stm32g474_core/board.cmake [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# keep first
+board_runner_args(stm32cubeprogrammer "--port=swd" "--reset-mode=hw")
```
[the remark this answers is not in the document]
Comment (ankayca) [reply boards/weact/stm32g474_core/board.cmake:4]: No it doesnt. let me correct

[the remark this answers is not in the document]
Comment (ankayca) [reply boards/weact/stm32g474_core/board.cmake]: Let me ask a question. The board doesn't have an embedded ST-link but I am using an external ST-link as a programmer. If the proper way to declare programmer is [DFU (this board also doesn't have an embedded ST-link)](https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/weact/stm32f405_core/board.cmake) , then I get this error and resetting device to boot mode not working to flashing with ST link.
`-- west flash: using runner dfu-util`
`Please reset your board to switch to DFU mode...`

So, how can I specify the programmer in my project file to keep board.cmake like the above board.cmake in DFU mode ?

**Comment by erwango** [comment]
Your pull request does not conform with Zephyr contribution guidelines. For reference, I'd advise you to check:
- [Contribution guidelines on docs.zephyrproject.org](https://docs.zephyrproject.org/latest/contribute/guidelines.html)
- [Great presentation dealing with "the best way to get your patches merged"](https://youtu.be/mnVNLM7ynMQ) and the related [Slides](https://static.sched.com/hosted_files/zephyr2022/a3/HOWTO_Get_Your_Zephyr_Patches_Merged_ZDS_2022.pdf)
- Coding Guidelines bot results reported below.

Please:
- Check your commit message. It should conform to the following template:
```
<dir>: <dir>: <Title summarizing the change>

<A description useful to the reader>

Signed-off-by: foo <foo@bar.baz>
```
- Remove the merge commit(s)
- We don't merge fixup commits. Please squash all fixes in the corresponding commits
- Split commits into multiple commits when your change impacts multiple subystems
- ...

**Pull request closed without merging**